src/main/java/com/powsybl/openloadflow/OpenLoadFlowProvider.java

```diff
@@ -262,18 +267,32 @@ private LoadFlowResult runDc(Network network, LoadFlowParameters parameters, Ope
         LOGGER.info("Add ratio to lines with different nominal voltage at both ends: {}", parametersExt.isAddRatioToLinesWithDifferentNominalVoltageAtBothEnds());
         LOGGER.info("Connected component mode: {}", parameters.getConnectedComponentMode());
 
-        DcLoadFlowParameters dcParameters = new DcLoadFlowParameters(slackBusSelector,
-                                                                     matrixFactory,
-                                                                     true,
-                                                                     parameters.isDcUseTransformerRatio(),
-                                                                     parameters.isDistributedSlack(),
-                                                                     parameters.getBalanceType(),
-                                                                     forcePhaseControlOffAndAddAngle1Var,
-                                                                     parametersExt.getPlausibleActivePowerLimit(),
-                                                                     parametersExt.isAddRatioToLinesWithDifferentNominalVoltageAtBothEnds(),
-                                                                     true,
-                                                                     parameters.getConnectedComponentMode() == LoadFlowParameters.ConnectedComponentMode.MAIN,
-                                                                     parameters.getCountriesToBalance());
+        var networkParameters = new LfNetworkParameters(slackBusSelector,
```

Comment:
Similarly, it might be worth it to put a `createDcLoadFlowParameters(parameters, parametersExt, slackBusSelector)`. And to factorize it with the other similar method (in DcSensitivityAnalysis)?

Reply:
Done but we cannot share the code as parameters are differents